Firestore/core/src/local/local_serializer.cc

```diff
@@ -453,6 +458,87 @@ BundledQuery LocalSerializer::DecodeBundledQuery(
       limit_type);
 }
 
+std::vector<model::Segment> LocalSerializer::DecodeFieldIndexSegments(
+    nanopb::Reader* reader, google_firestore_admin_v1_Index& index) const {
+  std::vector<model::Segment> result;
+  for (size_t i = 0; i < index.fields_count; ++i) {
+    const auto& field = index.fields[i];
+
+    util::StatusOr<FieldPath> field_path =
+        FieldPath::FromServerFormat(nanopb::MakeString(field.field_path));
+    if (!field_path.ok()) {
+      reader->Fail(
+          StringFormat("Failed to read field path for index segment: %s",
+                       nanopb::MakeString(field.field_path)));
+      return {};
+    }
+
+    Segment::Kind kind = Segment::Kind::kContains;
+    if (field.which_value_mode ==
+        google_firestore_admin_v1_Index_IndexField_order_tag) {
+      if (field.order ==
+          google_firestore_admin_v1_Index_IndexField_Order_ASCENDING) {
+        kind = Segment::Kind::kAscending;
+      } else {
+        kind = Segment::Kind::kDescending;
+      }
+    }
+
+    result.push_back({field_path.ValueOrDie(), kind});
+  }
+
+  return result;
+}
+
+nanopb::Message<google_firestore_admin_v1_Index>
+LocalSerializer::EncodeFieldIndexSegments(
+    const std::vector<model::Segment>& segments) const {
+  Message<google_firestore_admin_v1_Index> result;
+
+  result->query_scope =
+      google_firestore_admin_v1_Index_QueryScope_COLLECTION_GROUP;
+
+  result->fields_count = segments.size();
+  result->fields =
+      MakeArray<google_firestore_admin_v1_Index_IndexField>(segments.size());
```

Comment:
Looks like these new lines caused a few warnings. Reported in https://github.com/firebase/firebase-ios-sdk/issues/9430

Reply:
Thank you for reporting this. #9433 fixes the warnings.